[staging] shaderc: 2024.0 -> 2025.2, koboldcpp: 1.86.2 -> 1.92

4evy opened the pull request:
## Things done

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.92
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.86.2...v1.92

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

DontEatOreo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 401421`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>koboldcpp</li>
  </ul>
</details>

4evy committed:
shaderc: 2024.0 -> 2025.2

Diff: https://github.com/google/shaderc/compare/v2024.0...v2025.2

4evy committed:
shaderc: refactor

4evy committed:
koboldcpp: 1.86.2 -> 1.92

Changelog: https://github.com/LostRuins/koboldcpp/releases/tag/v1.92
Diff: https://github.com/LostRuins/koboldcpp/compare/v1.86.2...v1.92

4evy committed:
koboldcpp: enable `vulkanSupport` for darwin

DontEatOreo commented:
@ofborg eval

DontEatOreo commented:
`x86_64-linux` checks passed, and `aarch64-darwin` https://github.com/NixOS/nixpkgs/pull/401421/checks?check_run_id=42859539302 should work. But, Ofborg cut the LLVM compilation short, causing all dependencies to fail.

`aarch64-linux` https://github.com/NixOS/nixpkgs/pull/401421/checks?check_run_id=42845979699 also failed, specifically with the Nix daemon _(unrelated to `koboldcpp` or `shaderc`)_. But, similar to `aarch64-darwin`, it should work
